woocommerce/woocommerce-gateway-stripe
# Hide secret key and webhook secret

Fixes #2508 

## Changes proposed in this Pull Request:
The Stripe secret keys should not be visible in the UI as it can give a user access to make changes on the Stripe account. This PR replaces the full keys with masked versions in the frontend.

Additional context here: pc4etw-X0-p2

## Testing instructions
1. Go to **WooCommerce > Settings > Payments > Stripe > Settings** and click `Edit account keys`.
2. Verify that the keys are masked:
    ![Screenshot 2024-02-15 at 23 52 26](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/0ae68b83-ffda-49fb-8789-974fe114423a)
3. Save changes without modifying any values
4. Verify in the database that the keys were saved correctly (no masking):
    `SELECT * FROM wp_options WHERE option_name = 'woocommerce_stripe_settings'`
5. Click `Edit account keys`
6. Change the `publishable key` to an invalid value like `pk_test_INVALID VALUE`
7. Click the `Test connection` link in the bottom left corner
8. Check that the connection error message is displayed:
    ![Screenshot 2024-02-15 at 23 59 39](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/9fabf487-ece7-4292-b94e-80b4ff1e25d9)
9. Change the `publishable key` to a valid value (from your [Stripe dashboard](https://dashboard.stripe.com/test/apikeys))
10. Click the `Test connection` link in the bottom left corner
11. Check that the connection error message is displayed:
     ![Screenshot 2024-02-16 at 00 01 22](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/8fab85bd-06e3-4943-96ca-712598ad6b9d)
12. Click `Save test keys`
13. Verify in the database that the keys were saved correctly (no masking):
    `SELECT * FROM wp_options WHERE option_name = 'woocommerce_stripe_settings'`
